python/ray/data/_internal/execution/legacy_compat.py

```diff
@@ -147,7 +147,7 @@ def _blocks_to_input_buffer(blocks: BlockList, owns_blocks: bool) -> PhysicalOpe
             for b in i.blocks:
                 trace_allocation(b[0], "legacy_compat.blocks_to_input_buf[0]")
 
-        def do_read(blocks: Iterator[Block]) -> Iterator[Block]:
+        def do_read(blocks: Iterator[Block], task_idx: int) -> Iterator[Block]:
```

Comment:
Could we define a TaskContext class and put the index there? Otherwise, we may have to keep adding fields over time and it will get kind of messy.